lib/modules/auth/login/login_screen.dart

```diff
@@ -1,232 +1,210 @@
-    ScaffoldMessenger.of(context).showSnackBar(
-      SnackBar(
-        content: Text("$errorMessage"),
-      ),
-    );
-  }
 
-  void navigateToRegisterScreen(BuildContext context) async {
-    await Navigator.of(context).push(MaterialPageRoute(builder: (context) {
-      return RegisterScreen();
-    }));
-  }
+
+
+  ///////////////////////////////////////////////////////////
+  //////////////////// Widget methods ///////////////////////
+  ///////////////////////////////////////////////////////////
 
   Widget buildLoadingState() {
     return Center(
       child: CircularProgressIndicator(),
     );
   }
 
-  void buildToastMessage(String message, Color color) {
-    Fluttertoast.showToast(
-        msg: "$message",
-        toastLength: Toast.LENGTH_LONG,
-        gravity: ToastGravity.BOTTOM,
-        timeInSecForIosWeb: 5,
-        backgroundColor: color,
-        textColor: Colors.white,
-        fontSize: 16.0);
-  }
-}
+  Widget buildSizedBoxSeperator() => SizedBox(height: 16);
 
-/*Container buildLoginButton() {
-    return Container(
-      height: 45,
-      width: double.infinity,
-      child: ElevatedButton(
-        onPressed: () {},
-        child: Text("LOGIN"),
-      ),
-    );
-  }*/
-
-/*Widget buildEmailTextFormField() {
+  Widget buildEmailTextField() {
     return TextFormField(
+      textInputAction: TextInputAction.next,
       controller: emailController,
-      validator: (String value) {
-        if (value.isEmpty) {
-          return "Email Shouldn't be Empty";
-        } else
-          return null;
-      },
       keyboardType: TextInputType.emailAddress,
       decoration: InputDecoration(
         labelText: "Email",
+        errorText: emailErrorMessage,
         prefixIcon: Icon(Icons.email_outlined),
         border: OutlineInputBorder(
           borderRadius: BorderRadius.circular(20),
         ),
       ),
     );
-  }*/
+  }
 
-/*
-  Widget buildPasswordTextFormField() {
+  Widget buildPasswordTextField(BuildContext context) {
     return TextFormField(
+      textInputAction: TextInputAction.done,
       keyboardType: TextInputType.text,
-      obscureText: true,
+      obscureText: LoginBloc.get(context).isPassword,
       controller: passwordController,
-      validator: (String value) {
-        if (value.isEmpty) {
-          return "Password shouldn't be Empty";
-        } else
-          return null;
-      },
       decoration: InputDecoration(
         labelText: "Password",
+        errorText: passwordErrorMessage,
         prefixIcon: Icon(Icons.lock_outline_rounded),
+        suffixIcon: IconButton(
+          icon: Icon(LoginBloc.get(context).suffix),
+          onPressed: () {
+            changePasswordIconVisibility(context);
+          },
+        ),
         border: OutlineInputBorder(
           borderRadius: BorderRadius.circular(20),
         ),
       ),
     );
   }
-*/
+
+  Widget buildSignInButton(LoginState state) {
+    return ConditionalBuilder(
```

Comment:
why use a package for this (conditionalBuilder)

Reply:
done removing it